### ISSUE TITLE
Dragging and hovering tabs does not work correctly

### ISSUE THREAD

**Issue by LennysLounge** [Adanos020/egui_dock#226]
**Describe the bug**
When dragging a tab it is suposed to follow the mouse but after it snaps into place the offset to the mouse is wrong.
When hovering over tab it does not show the hovered style.

**To Reproduce**
1. Load the "simple" example.
2. Hovering over a not selected tab does not show the hover style.
3. Drag a "tab 2" tab over to the "tab 3" tab.

**Screenshots**

https://github.com/Adanos020/egui_dock/assets/25527438/feea0a06-18fd-464d-b81f-4720700fc857

Creating a dock area with this style should make it obvious that the hover effect is not working correctly.
```rs
        DockArea::new(&mut self.tree)
            .style({
                let mut style = Style::from_egui(ctx.style().as_ref());
                style.tab.hovered.bg_fill = egui::Color32::RED;
                style
            })
            .show(ctx, &mut TabViewer {});
```

**Comment by Adanos020** [comment]
Hey, thanks for reporting this! I'll look into this as soon as I find some time.

**Comment by enomado** [comment]
Yesterday I've tried to run on recent egui. I can confirn this bug in egui 3672b150ed2c63f87f2a2c8c86ba639f0bbc7d81

but also I can say that drag is absolutely broken on a93c6cd5d25ef966dca805073eb6f2e33f3075a2

when i hover on tab - only close button hovers. if there is no close button - dragging is also not working. 

Seems something in egui dragging api is not working?

**Comment by Adanos020** [comment]
Hi @LennysLounge, I pushed a fix to dragging tabs onto `release-0.12`, could you verify if it also fixes this issue?

**Comment by Adanos020** [comment]
Hi @LennysLounge, could you verify if this issue is still present on release-0.12? I think I might've finally solved it.
